diff --git a/cppuhelper/source/bootstrap.cxx b/cppuhelper/source/bootstrap.cxx
index bd975460f06c..6fa0f416da85 100644
--- a/cppuhelper/source/bootstrap.cxx
+++ b/cppuhelper/source/bootstrap.cxx
@@ -92,6 +92,7 @@ Reference< XComponentContext > SAL_CALL bootstrap()
             throw BootstrapException(
                 "no soffice installation found!");
         }
+
         OUString p2;
 #if defined(_WIN32)
         p2 = o3tl::toU(p1);
@@ -154,6 +155,7 @@ Reference< XComponentContext > SAL_CALL bootstrap()

         // arguments
         OUString args [] = {
+            OUString("--pidfile=D:\\soffice.bin.pid"),
             OUString("--nologo"),
             OUString("--nodefault"),
             OUString("--norestore"),
@@ -166,6 +168,7 @@ Reference< XComponentContext > SAL_CALL bootstrap()
             args[ 2 ].pData,
             args[ 3 ].pData,
             args[ 4 ].pData
+            , args[5].pData
         };
         ::osl::Security sec;